### PR TITLE
feat(dumps): drop dump V1 support

### LINKED ISSUE #1972: Drop support for dump V1
@meilisearch/docs-team and @meilisearch/integration-team (for the migration script).

Validated with @gmourier, no spec are needed here.

------

MeiliSearch > 0.25 will drop support for dumps version 1. This concerns MeiliSearch with versions before v0.21 (v0.21 not included). 

Meilisearch has evolved a lot since then, and it is becoming harder and harder to maintain non-breaking dumps over so many versions.

This is part of a process of stabilization, where eventually dumps will not break anymore.

If you need to import a dump for a version < v0.21, you will first need to import it to a version between v0.21 and v0.25, and re-export it to newer engine versions.

Ex: You are using MeiliSearch v0.20.0 and you want to upgrade to MeiliSearch v0.25.0. Here are the steps:
- Create a dump using MeiliSearch v0.20.0
- Start MeiliSearch v0.24.0 by importing the previous dump
- Create a dump using MeiliSearch v0.24.0 you just have launched
- Start MeiliSearch v0.25.0 by importing the dump created using v0.24.0

What will happen if you try to import a dump created in v0.20.0 into MeiliSearch v0.25.0 and later:

Capture d’écran 2021-12-07 à 10 41 45

### PR DESCRIPTION
**Pull request by MarinPostma**
Drop dump v1 support, and return an error message when trying to import a dump v1.

close #1972

### CODE REVIEW

File: meilisearch-lib/src/index_controller/dump_actor/loaders/v1.rs [hunk 1]
```diff
@@ -34,208 +20,5 @@ impl MetadataV1 {
...
-
-    #[test]
-    fn settings_format_regression() {
-        let settings = Settings::default();
-        assert_eq!(
-            r##"{"rankingRules":null,"distinctAttribute":null,"searchableAttributes":null,"displayedAttributes":null,"stopWords":null,"synonyms":null,"attributesForFaceting":null}"##,
-            serde_json::to_string(&settings).unwrap()
-        );
-    }
+        anyhow::bail!("The version 1 of the dumps is not supported anymore. You can re-export your dump from a version between 0.21 and 0.24, or start fresh from a version 0.25 onwards.")
```
Comment (curquiza) [inline meilisearch-lib/src/index_controller/dump_actor/loaders/v1.rs:23]: The `0.25` will stay even if the user tries to pass to v0.26.0 or later. Can we dynamically change this part, or re-do the sentence?

**Review by curquiza — CHANGES_REQUESTED** [review]

File: meilisearch-lib/src/index_controller/dump_actor/loaders/v1.rs (hunk 1 above)
Comment (MarinPostma) [reply meilisearch-lib/src/index_controller/dump_actor/loaders/v1.rs:23]: the support has been dropped in 0.25, hence "0.25 onward"

**Comment by curquiza** [comment]
A tiny detail, not sure this is important, but it seems like there is a part that failed here, is it expected?

Capture d’écran 2021-12-07 à 10 45 36

File: meilisearch-lib/src/index_controller/dump_actor/loaders/v1.rs (hunk 1 above)
Comment (MarinPostma) [reply meilisearch-lib/src/index_controller/dump_actor/loaders/v1.rs:23]: what do you want to convey @curquiza ?

Comment (curquiza) [reply meilisearch-lib/src/index_controller/dump_actor/loaders/v1.rs:23]: oh yes sorry, I completely missed the onwards! my bad!

**Comment by MarinPostma** [comment]
Uhm yes, this was already like this before. There should be a dump date here, but the date was only introduced in v2, so I think this was done to look like it was simple loading a dump.

**Review by curquiza — APPROVED** [review]
💪

**Review by irevoire — APPROVED** [review]

**Comment by curquiza** [comment]
bors merge

**Comment by curquiza** [comment]
bors merge

**Comment by curquiza** [comment]
I merge manually since bors is broken and we need to release :)

**Pull request merged**